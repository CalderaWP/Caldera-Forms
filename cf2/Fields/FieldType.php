<?php


namespace calderawp\calderaforms\cf2\Fields;


abstract class FieldType implements FieldTypeContract
{

	/** @inheritdoc */
	public static function toArray ()
	{
		return [
			'cf2' => TRUE,
			'field' => static::getName(),
			'description' => static::getDescription(),
			'category' => static::getCategory(),
			'setup' => static::getSetup(),
			'icon' => static ::getIcon(),
		];
	}

	/** @inheritdoc */
	public static function getSetup ()
	{
		return [
			'template' => 'fields/' . static::getCf1Identifier() . '/config.php',
			'preview' => 'fields/' . static::getCf1Identifier() . '/preview.php',
		];
	}

	/** @inheritdoc */
	public static function getIcon()
	{
		return caldera_forms_get_v2_container()->getCoreUrl() . 'assets/build/images/field.png';
	}
}
