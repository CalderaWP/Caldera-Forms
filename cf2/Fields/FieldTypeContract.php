<?php


namespace calderawp\calderaforms\cf2\Fields;


interface FieldTypeContract
{
    /**
     * Get the field's type
     *
     * @since 1.8.0
     *
     * @return string
     */
    public static function getType();

    /**
     * Get the field's identifier for use in cf1
     *
     * @since 1.8.0
     *
     * @return string
     */
    public static function getCf1Identifier();

	/**
	 * Get the field's setup array
	 *
	 * @since 1.8.0
	 *
	 * @return array
	 */
    public static function getSetup();

	/**
	 * Get the field's category
	 *
	 * @since 1.8.0
	 *
	 * @return string
	 */
    public static function getCategory();

	/**
	 * Get the field's description
	 *
	 * @since 1.8.0
	 *
	 * @return string
	 */
    public static function getDescription();

	/**
	 * Get the field's name
	 *
	 * @since 1.8.0
	 *
	 * @return string
	 */
    public static function getName();

	/**
	 * Convert to an array
	 *
	 * @since 1.8.0
	 *
	 * @return array
	 */
    public static function toArray();


	/**
	 * Get field type icon
	 *
	 * @since 1.8.0
	 *
	 * @return string
	 */
    public static function getIcon();
}
