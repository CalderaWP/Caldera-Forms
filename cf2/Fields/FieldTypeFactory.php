<?php


namespace calderawp\calderaforms\cf2\Fields;


use calderawp\calderaforms\cf2\Exceptions\NotFoundInContainerException;
use calderawp\calderaforms\cf2\Fields\FieldTypes\FileFieldType;
use Psr\Container\ContainerInterface;

class FieldTypeFactory implements FieldTypeFactoryContract
{
	/**
	 * The fields
	 *
	 * @since 1.8.0
	 *
	 * @var array
	 */
	protected $fields;

	/** @inheritdoc */
	public function add(FieldTypeContract $fieldType)
	{
		if (! $this->has(FileFieldType::getCf1Identifier())) {
			$this->fields[ $fieldType::getCf1Identifier() ] = $fieldType;
		}
		return $this;

	}

	/** @inheritdoc */
	public function getAll(){
		if( is_array( $this->fields ) ){
			return $this->fields;
		}

		return [];
	}


	/** @inheritdoc */
	public function get($id){
		if( ! $this->has( $id ) ){
			throw new NotFoundInContainerException(__( 'Field type not found', 'caldera-forms'),'404');
		}
		return $this->fields[$id];
	}

	/** @inheritdoc */
	public function has($id){
		return array_key_exists( $id,$this->getAll());
	}



}
