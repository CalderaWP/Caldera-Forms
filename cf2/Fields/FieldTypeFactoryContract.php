<?php


namespace calderawp\calderaforms\cf2\Fields;


use Psr\Container\ContainerInterface;

interface FieldTypeFactoryContract extends ContainerInterface
{
	/**
	 * Add a a field type
	 *
	 * @since 1.8.0
	 *
	 * @param FieldTypeContract $fieldType
	 *
	 * @return $this
	 */
	public function add(FieldTypeContract $fieldType);

	/**
	 * Get all fields
	 *
	 * @since 1.8.0
	 *
	 * @return array
	 */
	public function getAll();
}
