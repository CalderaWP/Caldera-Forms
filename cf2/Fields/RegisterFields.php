<?php


namespace calderawp\calderaforms\cf2\Fields;


class RegisterFields implements RegisterFieldsContract
{

	/**
	 * @var FieldTypeFactoryContract
	 */
	protected $factory;

	/**
	 * @var string
	 */
	protected $coreDirPath;

	/**
	 * RegisterFields constructor.
	 *
	 * @since 1.8.0
	 *
	 * @param FieldTypeFactoryContract $factory
	 * @param string $coreDirPath The path to the main plugin file
	 */
	public function __construct (FieldTypeFactoryContract $factory,$coreDirPath )
	{
		$this->factory = $factory;
		$this->coreDirPath = $coreDirPath;
	}

	/** @inheritdoc */
	public function getCoreDirPath()
	{
		return $this->coreDirPath;
	}

	/** @inheritdoc */
	public function filter($fields){
		if( ! empty( $this->factory->getAll() ) ){
			/** @var FieldType $field */
			foreach ( $this->factory->getAll() as $field ){
				$config = $field::toArray();
				foreach ( $config['setup' ] as $setUpKey => $setupItem ){
					$config['setup'][$setUpKey] = $this->getCoreDirPath() . $config['setup'][$setUpKey];
				}
				$config['cf2' ] = true;
				$fields[ $field::getCf1Identifier() ] = $config;
			}


		}
		return $fields;
	}


}
