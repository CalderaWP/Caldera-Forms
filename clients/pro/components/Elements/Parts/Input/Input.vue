<template>
	<input
			v-bind:type="type"
			v-bind:id="idAttr"
			@keyup="handle()"
			@change="handle()"
	/>
	<!--			v-model="value"-->
</template>
<script>
	export default{
		props: {
			idAttr: {
				type: String,
				require: true
			},
			value:{
				type:String,
				required: true
			},
			setting:{
				type:String,
				required: true
			},
			type: {
				type: String,
				default: 'text'
			}
		},
		methods: {
			handle(){
				this.$store.commit(this.setting,this.value);
			}
		},


	}
</script>