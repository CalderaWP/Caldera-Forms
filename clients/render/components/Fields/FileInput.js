import {CalderaFormsFieldGroup, Fragment} from "../CalderaFormsFieldGroup";
import {CalderaFormsFieldPropType} from "../CalderaFormsFieldRender";
import React from 'react';

import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';

/**
 * A FileInput
 *
 * @since 1.8.0
 *
 * @param props
 * @return {*}
 * @constructor
 */
export const FileInput = (props) => {

	const {onChange, accept, field, describedById, style, className, multiUploadText, multiple, inputProps, usePreviews, previewHeight, previewWidth, strings} = props;
	let {shouldDisable} = props;
	const {
		outterIdAttr,
		fieldId,
		fieldLabel,
		fieldCaption,
		required,
		isRequired,
		fieldPlaceHolder,
		fieldDefault,
		fieldIdAttr,
		isInvalid,
		fieldValue
	} = field;

	const valueSet = fieldValue.length > 0;
	const removeFileID = fieldIdAttr + '_file_';
	const buttonControls = fieldIdAttr + ', cf2-list-files';

	inputProps.id = fieldIdAttr;

	if (valueSet && !multiple) {
		shouldDisable = true;
		inputProps.disabled = true;
	}

	return (

		<div className="cf2-dropzone" data-field={fieldId}>
			<Dropzone
				onDrop={onChange}
				className={className}
				accept={'string' === typeof  accept ? accept : ''}
				style={style}
				disabled={shouldDisable}
				inputProps={inputProps}
				disableClick={shouldDisable}
				multiple={multiple}
			>
				<button
					type="button"
					className="btn btn-block"
					aria-controls={buttonControls}
					aria-expanded={valueSet}
					disabled={shouldDisable}
				>
					{multiUploadText}
				</button>
			</Dropzone>

			{valueSet &&
			<ul
				id="cf2-list-files"
				role="list"
			>
				{
					fieldValue.map(
						(file, index) =>
							<li
								id={removeFileID + index}
								key={index}
								className="cf2-file-listed"
								role="listitem"
								aria-posinset={index}
							>

								<button
									type="button"
									aria-controls={removeFileID + index}
									data-file={removeFileID + index}
									className="cf2-file-remove"
									onClick={(e) => onChange(e, file)}
								>
									<span className="screen-reader-text sr-text">{strings.removeFile}</span>
								</button>

								<div>
									{usePreviews === true && file.type.startsWith("image") === true ?
										<img
											className="cf2-file-field-img-preview"
											width={previewWidth}
											height={previewHeight}
											src={file.preview}
											alt={file.name}
										/>
										:
										<span className="cf2-file-name">{file.name}</span>
									}
									<br/>
									<span
										className="cf2-file-data"> {file.type} - {file.size} bytes - {file.type.startsWith("image")}</span>
								</div>
							</li>
					)
				}
			</ul>
			}
		</div>

	)

};

/**
 * Prop definitions for FileInput component
 *
 * @since 1.8.0
 *
 * @type {{field: *, onChange: (e|*), shouldDisable: *, isInvalid: *, describedById: *, multiple: *, text: *, multiUploadText: *, message: *, style: *, previewStyle: *, inputProps: *, className: *, accept: *}}
 */
FileInput.propTypes = {
	field: PropTypes.shape(CalderaFormsFieldPropType),
	onChange: PropTypes.func,
	shouldDisable: PropTypes.bool,
	isInvalid: PropTypes.bool,
	describedById: PropTypes.string,
	multiple: PropTypes.oneOfType([
		PropTypes.bool,
		PropTypes.number
	]),
	text: PropTypes.object,
	multiUploadText: PropTypes.oneOfType([
		PropTypes.bool,
		PropTypes.string
	]),
	message: PropTypes.shape({
		error: PropTypes.bool,
		message: PropTypes.string
	}),
	style: PropTypes.object,
	usePreviews: PropTypes.oneOfType([
		PropTypes.bool,
		PropTypes.string
	]),
	previewWidth: PropTypes.number,
	previewHeight: PropTypes.number,
	inputProps: PropTypes.object,
	className: PropTypes.string,
	accept: PropTypes.oneOfType([
		PropTypes.bool,
		PropTypes.string
	]),
	isRequired: PropTypes.oneOfType([
		PropTypes.bool,
		PropTypes.object
	]),
};

/**
 * Default props for a FileInput component
 *
 * @since 1.8.0
 *
 * @type {{message: {error: boolean, message: string}, text: {buttonText: string, removeFile: string}, multiUploadText: string, inputProps: {type: string}, disableClick: boolean, multiple: boolean, className: string, previewStyle: {height: string, width: string}}}
 */
FileInput.defaultProps = {
	message: {
		error: false,
		message: ''
	},
	inputProps: {
		type: 'file',
		disabled: 'false'
	},
	disableClick: false,
	className: 'cf2-file form-control'
};

/**
 * Prepare a field's config to be used as props for FileInput component
 *
 * @since 1.8.0
 *
 * @param {*} fieldConfig The field's configuration
 *
 * @return {{field: *}}
 */
FileInput.fieldConfigToProps = (fieldConfig) => {
	let props = {
		field: fieldConfig
	};
	const configOptionProps = [
		'multiple',
		'multiUploadText'
	];

	if (!props.field.hasOwnProperty('isRequired')) {
		props.field.isRequired = false;
	}
	if (fieldConfig.hasOwnProperty('configOptions')) {

		const {configOptions} = fieldConfig;

		configOptionProps.forEach(key => {
			props[key] = configOptions[key];
		});

		if (configOptions.hasOwnProperty('allowedTypes')) {

			props.inputProps = {
				type: 'file',
				accept: configOptions.allowedTypes
			};
			props.accept = configOptions.allowedTypes;

		} else {
			props.accept = '';
		}

		if (fieldConfig.configOptions.hasOwnProperty('multiple')) {
			if (fieldConfig.configOptions.multiple === 1) {
				props.multiple = true;
			} else {
				props.multiple = false;
			}
		} else {
			props.multiple = false;
		}

		if (configOptions.hasOwnProperty('usePreviews')) {

			if (configOptions.usePreviews === true) {
				props.usePreviews = true;
				props.previewWidth = configOptions.previewWidth;
				props.previewHeight = configOptions.previewHeight;
			} else {
				props.usePreviews = false;
			}

		}


	}

	return props;
};
