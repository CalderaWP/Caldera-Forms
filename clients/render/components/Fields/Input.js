import {CalderaFormsFieldGroup, Fragment} from "../CalderaFormsFieldGroup";
import {CalderaFormsFieldPropType} from "../CalderaFormsFieldRender";
import PropTypes from 'prop-types';

/**
 * Input component
 *
 * @since 1.8.0
 *
 * @param props
 * @return {*}
 * @constructor
 */
export const Input = (props) => {
	const{shouldDisable,field,describedById,onChange} = props;
	const {
		type,
		outterIdAttr,
		fieldId,
		fieldLabel,
		fieldCaption,
		required,
		fieldPlaceHolder,
		fieldDefault,
		fieldValue,
		fieldIdAttr,
		isInvalid
	} = field;


	const propsThatDoNotUseAnEquals = {};
	if (shouldDisable) {
		propsThatDoNotUseAnEquals.disabled = true;
	} else {
		propsThatDoNotUseAnEquals.disabled = false;
	}
	if (required) {
		propsThatDoNotUseAnEquals.required = true;
	} else {
		propsThatDoNotUseAnEquals.required = false;
	}

	return(
		<input
			type={type}
			{...propsThatDoNotUseAnEquals}
			aria-describedby={describedById}
			value={fieldValue}
			className={'cf2-text form-control'}
			id={fieldIdAttr}
			placeholder={fieldPlaceHolder}
			onChange={onChange}
			data-field={fieldId}
			data-type={'cf2-text'}
			name={fieldId}
		/>
	)
};

/**
 * Prop definitions for Input component
 *
 * @since 1.8.0
 *
 * @type {{field: *, onChange: (e|*), shouldDisable: *, isInvalid: *, describedById: *}}
 */
Input.propTypes = {
	field: PropTypes.shape(CalderaFormsFieldPropType),
	onChange: PropTypes.func.isRequired,
	shouldDisable: PropTypes.bool,
	isInvalid: PropTypes.bool,
	describedById: PropTypes.string,
};

/**
 * Default props for Input component
 *
 * @since 1.8.0
 *
 * @type {{ariaAttr: string, isInvalid: boolean}}
 */
Input.defaultProps = {
	ariaAttr: '',
	isInvalid:false
};
