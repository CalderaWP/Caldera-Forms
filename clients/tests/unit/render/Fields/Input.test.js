import renderer from 'react-test-renderer';
import{Input} from "../../../../render/components/Fields/Input";
describe( 'Input field', () => {
	const fieldIdAttr = 'fld_5843941_1';
	const fieldConfig = {
		fieldCaption: "",
		fieldDefault: "new default",
		fieldId: "fld_5843941",
		fieldIdAttr,
		fieldLabel: "Text 2 Field",
		fieldPlaceHolder: "",
		fieldValue: "",
		outterIdAttr: "cf2-fld_5843941_1",
		required: false,
		type: "text",
	};
	const handler = () => {
	};


	it( 'matches snapshot with basic args', () => {
		expect(renderer.create(
			<Input
				onChange={handler}
				shouldDisable={false}
				field={fieldConfig}
			/>
			).toJSON()
		).toMatchSnapshot();
	});

	it( 'prints aria attribute', () => {

		expect(renderer.create(
			<Input
				onChange={handler}
				shouldDisable={false}
				field={fieldConfig}
				describedById={'foo'}
			/>
			).toJSON()
		).toMatchSnapshot();


	});

});