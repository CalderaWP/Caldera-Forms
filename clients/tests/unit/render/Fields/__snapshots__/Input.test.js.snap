// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field matches snapshot with basic args 1`] = `
<input
  className="cf2-text form-control"
  data-field="fld_5843941"
  data-type="cf2-text"
  disabled={false}
  id="fld_5843941_1"
  name="fld_5843941"
  onChange={[Function]}
  placeholder=""
  required={false}
  type="text"
  value=""
/>
`;

exports[`Input field prints aria attribute 1`] = `
<input
  aria-describedby="foo"
  className="cf2-text form-control"
  data-field="fld_5843941"
  data-type="cf2-text"
  disabled={false}
  id="fld_5843941_1"
  name="fld_5843941"
  onChange={[Function]}
  placeholder=""
  required={false}
  type="text"
  value=""
/>
`;
